nucypher/blockchain/eth/actors.py

```diff
@@ -483,29 +460,32 @@ def divide_stake(self,
         if additional_periods and expiration:
             raise ValueError("Pass the number of lock periods or an expiration MayaDT; not both.")
 
+        # Update staking cache element
+        stakes = self.stakes
+
         # Select stake to divide from local cache
         try:
-            current_stake = self.stakes[stake_index]
+            current_stake = stakes[stake_index]
         except KeyError:
-            if len(self.stakes):
+            if len(stakes):
                 message = f"Cannot divide stake - No stake exists with index {stake_index}."
             else:
                 message = "Cannot divide stake - There are no active stakes."
             raise Stake.StakingError(message)
 
         # Calculate stake duration in periods
         if expiration:
-            additional_periods = datetime_to_period(datetime=expiration) - current_stake.end_period
+            additional_periods = datetime_to_period(datetime=expiration) - current_stake.last_locked_period
```

Comment:
is last_locked_period better than end_period? Could be interpreted as the most recent period. Maybe final_locked_period?

Reply:
Possibly. I always have a problem with `end_something` because it's not clear if this `something` is actually included in the range or not. So either `final`, or `last` sounds good (but I agree that final is better)

Reply:
:heavy_check_mark: